app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt

```diff
@@ -552,10 +552,18 @@ class BrowserFragment : Fragment(), BackHandler, CoroutineScope {
             readerViewFeature.onBackPressed() -> true
             customTabsIntegration.onBackPressed() -> true
             sessionFeature.onBackPressed() -> true
-            else -> false
+            else -> {
+                removeSessionIfNeeded()
+                false
+            }
         }
     }
 
+    private fun removeSessionIfNeeded() {
+        val session = getSessionById() ?: return
+        if (session.source == Session.Source.ACTION_VIEW) requireComponents.core.sessionManager.remove(session)
```

Comment:
Should this function return true if it handles the back press by removing the session?

Reply:
Sorry about the drive-by comment, I came here to actually report this same bug!

Reply:
I don't think so because I just want to do this as an extra when all other features don't respond to the backpress and let the system go back to the previous app. See implementation in RB here: https://github.com/mozilla-mobile/reference-browser/blob/master/app/src/main/java/org/mozilla/reference/browser/BrowserActivity.kt#L77

Reply:
Ah gotcha, this was code that I think needed to go into the session feature. We can refactor that later.

Reply:
I definitely think it should go into the session feature because I'm not sure if Fenix on it's own right now can implement the second requested part of this feature "transfer the session to a normal session if a user navigates on it". I think we'll need to remove the source flag which we can't currently do on our side.